### PR TITLE
Add --debug to open a query in ActiveData's query tool #17

### PR DESCRIPTION
**Pull request by TrangEmily**
Thank @klahnakoski for detailed information. I followed it and made a modification. And I have some concerns: 1) When I tried to run the query, the command is "adr query" instead of "adr-query"; 2) I used `FAKE_CONTEXT["debug"] = args.debug`; but not sure it is a good solution or not 3) When opening a browser, I tried to use args.url, it stores "http://activedata.allizom.org/query" (which is the same with the url in the guide) but it didn't work, so I tried "http://activedata.allizom.org/tools/query.html" instead. Thank you.

### CODE REVIEW

**Comment by ahal** [comment]
> 1) When I tried to run the query, the command is "adr query" instead of "adr-query";

This is fine, a commit recently landed that changed this. So you are correct, it's now called `adr query`.

> 2) I used `FAKE_CONTEXT["debug"] = args.debug`; but not sure it is a good solution or not

I don't think you need to put 'debug' in the context, it should be irrelevant to the queries that are being run.

> 3) When opening a browser, I tried to use args.url, it stores "http://activedata.allizom.org/query" (which is the same with the url in the guide) but it didn't work, so I tried "http://activedata.allizom.org/tools/query.html" instead. Thank you.

Yes, that is the correct URL

Thanks for your PR! I'll try to take a look soon and I'll let you know if there are any changes that need to be made.

**Comment by TrangEmily** [comment]
Thank you very much @ahal. Please let me know what I should change/improve.

File: adr/cli.py [hunk 1]
```diff
@@ -43,7 +44,11 @@ def query_handler(args, remainder):
         if query not in queries:
             log.error("query '{}' not found!".format(query))
             continue
-        print(format_query(query, args, fmt=args.fmt))
+        data, query_id = format_query(query, args, fmt=args.fmt)
+        print(data)
```
Comment (ahal) [inline adr/cli.py]: nit: you missed removing a debug statement

File: adr/cli.py [hunk 2]
```diff
@@ -43,7 +44,11 @@ def query_handler(args, remainder):
...
+        if query_id:
+            url = "http://activedata.allizom.org/tools/query.html#query_id={}".format(query_id)
```
Comment (ahal) [inline adr/cli.py]: This should be based on the `ACTIVE_DATA_URL` (so if someone specifies --url, then the query url will also be updated). You can put it in a similar global in `query.py`, e.g:

    ACTIVE_DATA_QUERY_URL = "{}/tools/query.html".format(ACTIVE_DATA_URL)

Then you can append the `#query_url=<id>` part back in `cli.py` where you have it currently.

File: adr/query.py [hunk 3]
```diff
@@ -112,8 +114,13 @@ def format_query(query, args, fmt='table'):
     if isinstance(fmt, string_types):
         fmt = all_formatters[fmt]
 
+    FAKE_CONTEXT["debug"] = args.debug
```
Comment (ahal) [inline adr/query.py]: Please remove this.

**Comment by TrangEmily** [comment]
Thank you @ahal I updated the code: 1) Add a 'time.sleep(2)' into query.py 2) Use `urlparse(ACTIVE_DATA_URL)` to get the url 3) Add a parameter in this method `def run_query(name, debug=False, **context):` instead of using context. Please help me to have a look and tell me if I should change. Thank you.

File: adr/query.py [hunk 4]
```diff
@@ -8,13 +8,18 @@
 import requests
 import yaml
 from six import string_types
-
 from adr.formatter import all_formatters
+import sys
+if sys.version_info >= (3, 0):
```
[the remark this answers is not in the document]
Comment (TrangEmily) [reply adr/query.py]: Thank you, I updated the code and committed.

**Comment by ahal** [comment]
Thanks for your contribution!

**Pull request merged**